marimo/_save/save.py

```diff
@@ -511,3 +364,289 @@ def __exit__(
+
+    1. `mo.cache` persists its cache even if the cell defining the
+        cached function is re-run, as long as the code defining the function
+        (excluding comments and formatting) has not changed.
+    2. `mo.cache` keys on closed-over values in addition to function arguments,
+        preventing accumulation of hidden state associated with
+        `functools.cache`.
+    3. `mo.cache` does not require its arguments to be
+        hashable (only pickleable), meaning it can work with lists, sets, NumPy
+        arrays, PyTorch tensors, and more.
+
+    `mo.cache` obtains these benefits at the cost of slightly higher overhead
+    than `functools.cache`, so it is best used for expensive functions.
+
+    Like `functools.cache`, `mo.cache` is thread-safe.
+
+    The cache has an unlimited maximum size. To limit the cache size, use
+    `@mo.lru_cache`. `mo.cache` is slightly faster than `mo.lru_cache`, but in
+    most applications the difference is negligible.
+
+    Note, `mo.cache` can also be used as a drop in replacement for context block
+    caching like `mo.persistent_cache`.
+
+    **Args**:
+
+    - `pin_modules`: if True, the cache will be invalidated if module versions
+      differ.
+    """
+    arg = name
+    del name
+
+    if loader is None:
+        loader = MemoryLoader.partial(max_size=-1)
+
+    return _cache_invocation(
+        arg,
+        loader,
+        *args,
+        frame_offset=_frame_offset + 1,
+        **kwargs,
+    )
+
+
+def lru_cache(
+    name: Union[str, Optional[Callable[..., Any]]] = None,
+    maxsize: int = 128,
+    *args: Any,
+    **kwargs: Any,
+) -> Union[_cache_call, _cache_context]:
+    """Decorator for LRU caching the return value of a function.
+
+    `mo.lru_cache` is a version of `mo.cache` with a bounded cache size. As an
+    LRU (Least Recently Used) cache, only the last used `maxsize` values are
+    retained, with the oldest values being discarded. For more information,
+    see the documentation of `mo.cache`.
+
+    **Usage.**
+
+    ```python
+    import marimo as mo
+
+
+    @mo.lru_cache
+    def factorial(n):
+        return n * factorial(n - 1) if n else 1
+    ```
+
+    **Args**:
+
+    - `maxsize`: the maximum number of entries in the cache; defaults to 128.
+      Setting to -1 disables cache limits.
+    - `pin_modules`: if True, the cache will be invalidated if module versions
+      differ.
+    """
+    arg = name
+    del name
+
+    if {"loader"} & set(kwargs.keys()):
+        raise ValueError(
+            "loader is not a valid argument "
+            "for lru_cache, use mo.cache instead."
+        )
+
+    return cache(
+        arg,
+        *args,
+        loader=MemoryLoader.partial(max_size=maxsize),
+        _frame_offset=2,
+        **kwargs,
+    )
+
+
+def persistent_cache(
+    name: Union[str, Optional[Callable[..., Any]]] = None,
+    save_path: str | None = None,
+    *args: Any,
+    **kwargs: Any,
```

Comment:
Same question, why varargs/kwargs?

Reply:
In addition to matching the signature throughout, extra `kwargs` are now used by Loaders to enable:

```python
with MyLoader.cache("my_cache", constructor_values=123) as c:
    ...
```

or

```python
@MyLoader.cache
def my_fn(args):
    ...
```

a bit of feature creep, but even when testing these changes, I noticed how cumbersome 

```python
with cache(_loader=MyLoader.partial(constructor_values=123)) as c:
    ...
```

was, and seemed like low hanging fruit.